practices/git_flow.md

```diff
@@ -0,0 +1,77 @@
+
+![Git Flow branching model](/files/git_flow.png)
+[nvie: A successful Git branching model](https://nvie.com/posts/a-successful-git-branching-model/)
+
+**The main branches:**
+- **Master branch** stores the official release history. It contains an abridged version of your project. It is the core of the branching model with an infinite lifetime.
+
+- **Develop branch** is an integration branch for features. It contains the complete history of your project. It also has an infinite lifetime. When the code in this branch is stable and ready for a release, the changes are merged to the master branch and tagged with a release number.
+
+**Supporting branches:**
+- **Feature branches** are created for each new feature. Their parent branch is the develop branch. Features should never interact directly with the master branch.
+
+- **Release branch** is created when there are enough features for a release. It enables one team to work on a current release while another team keeps working on features for the next release. Release branches’ parent is also the develop branch. This branch is where [continuous delivery](/practices/continuous-delivery) happens.
+
+- **Hotfix branches** are maintenance branches. They are created to quickly patch production release. These branches enable the team to fix mistakes without interrupting the workflow or waiting for the next release. Their parent is the master branch. Hotfix branches are the only branches to be forked directly off of the master branch. You can automatize the hotfix branch creation: whenever a mistake appears, the hotfix branch is automatically created.
+
+## Why You Might Want to Use the Git Flow Workflow
+- Git Flow Workflow simplifies parallel development because it isolates the new development from the released project. You can work on any project version. The new development is done in feature branches so it is merged back when the developer is satisfied with the code.
+- Your team can collaborate better and spends less time managing the project versions if they use simple and clear branching strategy, such as Git Flow Workflow.
+- You can always commit the changes and create a new feature branch when you need to interrupt your work. You can come back to your feature at any time.
+- Hotfix branches allow your team to make emergency changes. You do not have to worry that you would accidentally merge in a new development at the same time.
+
+## Issues the Git Flow Workflow Solves
+- Unhappy clients
+- [Poor code quality](/issues/poor-code-quality)
+- Toxic team culture
+- Knowledge hoarding
+- Not my problem mentality
+- Meaningless work
+
+## How to Implement the Git Flow Workflow
+There are lots of different applications for Git Flow Workflow (such as [SmartGit](https://www.syntevo.com/smartgit/) or [GitKraken](https://www.gitkraken.com/). Pick the application suitable for your operation system to initialize the Git Flow. Customize your project setup by following these steps:
```

Comment:
The most used is SourceTree, please add it.

Reply:
> `There are lots of different applications for Git Flow Workflow`

It's not about the application for Git Flow... You should say that all these Git applications supports Git Flow branching model and has a feature (specific settings) for that.